primitives/src/currency.rs

```diff
@@ -430,3 +430,9 @@ pub struct AssetMetadata<Balance> {
 	pub decimals: u8,
 	pub minimal_balance: Balance,
 }
+
+#[derive(Clone, Eq, PartialEq, RuntimeDebug, Encode, Decode, TypeInfo)]
+pub struct ForeignAssetExtraMetadata {
+	pub maybe_currency_id: Option<CurrencyId>,
+	pub maybe_location: Option<xcm::latest::MultiLocation>,
```

Comment:
rename to `override_currency_id` and `override_location`. IIUC, `maybe_currency_id` will override ForeignAsset(id), and `maybe_location` will override registered location using in the `AssetIdMapping` trait.

Reply:
Yep renamed